src/providers/treeViewProvider.ts

```diff
@@ -0,0 +1,63 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export class HttpTreeProvider implements vscode.TreeDataProvider<HttpClientItem> {
+
+    private _onDidChangeTreeData: vscode.EventEmitter<HttpClientItem | undefined> = new vscode.EventEmitter<HttpClientItem | undefined>();
+    readonly onDidChangeTreeData: vscode.Event<HttpClientItem | undefined> = this._onDidChangeTreeData.event;
+
+    constructor() {
+    }
+
+    refresh(): void {
+        this._onDidChangeTreeData.fire();
+    }
+
+    getTreeItem(element: HttpClientItem): vscode.TreeItem {
+        return element;
+    }
+
+    getChildren(): Thenable<HttpClientItem[]> {
+        return Promise.resolve(this.getDepsInPackageJson());
+    }
+
+    private getDepsInPackageJson(): Thenable<HttpClientItem[]> {
+        return vscode.workspace.findFiles("**/*.http").then((values: vscode.Uri[]) => {
+            return values.map((val: vscode.Uri) => {
```

Comment:
Personally I prefer the http request tree view corresponds to the active HTTP document in editor. That means the tree view displays all the requests in the current open file, as user change to other http files, the content in tree view will refresh.

Reply:
I don't always need to open the file, so I prefer to just run the request without changing the context of my code editor. So my suggestion is:
1. Make a preference if to just launch the request or to open the editor and launch the request.
2. If there is more then one request in the file, then show the requests as child items in the tree.

Reply:
> 1. Make a preference if to just launch the request or to open the editor and launch the request.

I think we can provide these both, like if click the request URL(or name for named request) in the tree view, it means open the editor, if click the icon in the tree item means sending the request.



> 2. If there is more then one request in the file, then show the requests as child items in the tree.

Personally I think always listing all the requests as the child items are acceptable.

Reply:
i have made the changes according to this discussion.